api/v1alpha1/deviceprofile_types.go

```diff
@@ -20,6 +20,10 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const (
+	DeviceProfileFinalizer = "v1alpha1.deviceProfile.finalizer"
```

Comment:
Changing finalizer is a break change, we should add some logic for compatibility purpose when deleting device profile.